### ISSUE TITLE
Customizable benchmark executor and reporter

### ISSUE THREAD

**Issue by KDr2** [TuringLang/ContinuousBenchmarks.jl#44]
- Provide a way to let the users customize how the benchmarks are executed and how/what results (CPU TIME, USE TIME, MEM ALLOCATION, and so on) should be collected
- Provide a method to let the users customize the report (content, format, etc.)

**Comment by yebai** [comment]
> Provide a way to let the users customize how the benchmarks are executed and how/what results (CPU TIME, USE TIME, MEM ALLOCATION, and so on) should be collected

I suggest that we replace our  `@tbenchmark` with the `@benchmark` (and `@btime`) macro from  [`BenchmarkTools.jl`](https://github.com/JuliaCI/BenchmarkTools.jl). We can also store the `BenchmarkTools.Trial` object instead of a `Dict` object for benchmarking results. 

> Provide a method to let the users customize the report (content, format, etc.)

Sounds great -- I suggest we store all the benchmarking results in a `DataFrame` and allow the user to overload `show(res::DataFrame)` to get customised reports.

**Comment by trappmartin** [comment]
This looks great! 

Here are a few suggestions:

1.) I think it might be good to be able to store some additional information: ESS, Results from convergence diagnostics (e.g. Rhat)
2.) I think it might be good if we separate benchmarking of Turing from benchmarking Turing against other PPLs. In my opinion we should have a separate Repo or so to run benchmarks agains other PPLs and should only do regression testing using the benchmark script & models in the Turing repo.

**Comment by KDr2** [comment]
Using DataFrame to store the result is really a good idea.

The users can customize the benchmark by giving different columns, and we can generate the report according to the column names in a unified way without any customization effort on the user side.

We can change the signature of the function `build_log_data` as follows:
```
using Turing, TuringBenchmarks.TuringTools

res3 = BenchmarkTools.@benchmark chain3 = TuringAlg()
res4 = BenchmarkTools.@benchmark chain4 = StanAlg()

report = build_log_data("benchmarkingName", columns=("Time", "Memory", "GC"...), TuringAlg=res3, StanAlg=res4)
print_log(LOG_DATA)
```

**Comment by KDr2** [comment]
Closed in favor of #46.
